server/chat-plugins/scavengers.ts

```diff
@@ -1468,88 +1731,104 @@ const commands = {
-		this.modlog('SCAV BLITZ', null, `${type}: ${blitzPoints}`);
+		this.privateModAction(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints}.)`);
+		this.modlog('SCAV BLITZ', null, `${gameType}: ${blitzPoints}`);
 
 		// double modnote in scavs room if it is a subroomgroupchat
-		if (room.parent && !room.chatRoomData) {
-			scavsRoom.modlog(`(scavengers) SCAV BLITZ: by ${user.id}: ${type}: ${blitzPoints}`);
-			scavsRoom.sendMods(`(${user.name} has set the points awarded for blitz for ${type} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
-			scavsRoom.roomlog(`(${user.name} has set the points awarded for blitz for ${type} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
+		if (room.parent && !room.chatRoomData && scavsRoom) {
+			scavsRoom.modlog(`(scavengers) SCAV BLITZ: by ${user.id}: ${gameType}: ${blitzPoints}`);
+			scavsRoom.sendMods(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
+			scavsRoom.roomlog(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
 		}
 	},
 
 	sethostpoints(target, room, user) {
-		if (room.roomid !== 'scavengers' && !(room.parent && room.parent.roomid === 'scavengers')) return this.errorReply("This command can only be used in the scavengers room.");
+		if (room.roomid !== 'scavengers' && !(room.parent && room.parent.roomid === 'scavengers')) {
+			return this.errorReply("This command can only be used in the scavengers room.");
+		}
 		if (!this.can('mute', null, room)) return false; // perms for viewing only
-		if (!target) return this.sendReply(`The points rewarded for hosting a regular hunt are ${Object.hasOwnProperty.call(room, 'hostPoints') ? room.hostPoints : DEFAULT_HOST_POINTS}`);
+		if (!target) {
+			const pointSetting = Object.hasOwnProperty.call(room.scavSettings, 'hostPoints') ?
+				room.scavSettings.hostPoints : DEFAULT_HOST_POINTS;
+			return this.sendReply(`The points rewarded for hosting a regular hunt is ${pointSetting}.`);
+		}
 
 		if (!this.can('declare', null, room)) return false; // perms for editting
 		const points = parseInt(target);
 		if (isNaN(points)) return this.errorReply(`${target} is not a valid number of points.`);
-		room.hostPoints = points;
+
+		room.scavSettings.hostPoints = points;
 		if (room.chatRoomData) {
-			room.chatRoomData.hostPoints = room.hostPoints;
+			room.chatRoomData.scavSettings = room.scavSettings;
 			Rooms.global.writeChatRoomData();
 		}
 		this.privateModAction(`(${user.name} has set the points awarded for hosting regular scavenger hunts to ${points})`);
-		this.modlog('SCAV SETHOSTPOINTS', null, points);
+		this.modlog('SCAV SETHOSTPOINTS', null, `${points}`);
 
 		// double modnote in scavs room if it is a subroomgroupchat
-		if (room.parent && !room.chatRoomData) {
+		if (room.parent && !room.chatRoomData && scavsRoom) {
```

Comment:
Don't you already have a `"This command can only be used in the scavengers room."` error up above? I would maybe combine those? Make a `getScavsRoom` function that errors if it doesn't exist.

Reply:
This was only for typescript to shut up since scavsRoom (Rooms.get()) could be null, and typescript complained.

Reply:
I know. I think it should check for it to be `null` up in the same place as the "can only be used in scavengers" error message, and it should be a combined check. I posted some code for that earlier.